Source/quests.cpp

```diff
@@ -505,7 +494,7 @@ void CheckQuestKill(const MonsterStruct &monster, bool sendmsg)
 void DRLG_CheckQuests(int x, int y)
 {
 	for (auto &quest : Quests) {
-		if (Quests[static_cast<quest_id>(quest._qidx)].IsAvailable()) {
+		if (Quests[quest._qidx].IsAvailable()) {

```

Comment:
```suggestion
		if (quest.IsAvailable()) 
```
No real reason to go back into the array here, _qidx is only equivalent to the index because of the way quest_ids are defined.